app/models/course_option.rb

```diff
@@ -12,12 +12,27 @@ class CourseOption < ApplicationRecord
   }
 
   enum vacancy_status: {
-    both_full_time_and_part_time_vacancies: 'both_full_time_and_part_time_vacancies',
-    part_time_vacancies: 'part_time_vacancies',
-    full_time_vacancies: 'full_time_vacancies',
+    vacancies: 'vacancies',
     no_vacancies: 'no_vacancies',
   }
 
+  def update_vacancy_status_from_detailed_description!(description)
```

Comment:
Could we move this into `SyncProviderFromFind` or a separate service, to avoid models from becoming too large?

Reply:
Counterpoints:

- Moving behavior like this into the service layer pushes us towards an [Anemic Domain Model](https://martinfowler.com/bliki/AnemicDomainModel.html).
- The method acts solely on the `vacancy_status` field on records in the `course_options` table, deciding how to translate and persist valid input values. This seems like an appropriate responsibility of the `CourseOption` active record, IMHO.
- The method contains a piece of domain logic, rather than doing the coordination and delegating work that service objects typically do according to their definition in DDD.

Alternatives:

- Identify a distinct domain concept and put this logic into a separate model. Something like a `FindApiVacancyDescription`, which instantiates with the string from Find and implements a `vacancy_status(study_mode:)` method.
- Leave the method as-is, refactoring if the model becomes unwieldy at a later time.

Reply:
Hah, you've thought about this :)

So it's worked out quite well to only add to the model if absolutely necessary. The models are easy to read, classes have few responsibilities and they're easily refactored.

In particular though, I think this method contains detailed knowledge about Find's domain model, which is outside of the responsibilities of the model. It will never be used outside of `SyncProviderFromFind`, so it seems quite clean to keep it in there. I see the translation of attributes from Find -> Apply no different than what we already do, like https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/blob/184c780884bbf8a0599118d9172cd297d11f6464/app/services/sync_provider_from_find.rb#L67-L74

I think our definitions of services is different from others, we've tended to use "service" as a name for all POROs that we have, not just things that do things.

Reply:
> It will never be used outside of SyncProviderFromFind, so it seems quite clean to keep it in there.

Fair point, I can see the value of co-locating this logic.

> I think our definitions of services is different from others, we've tended to use "service" as a name for all POROs that we have, not just things that do things.

Seems like another hint that the organization of that folder could do with a rethink at some point 😉 .

Will get an update pushed shortly.